src/operator/bilinear_sampler.cc

```diff
@@ -28,7 +28,7 @@
 namespace mshadow {
 template<typename DType>
 bool between(DType value, int lowerBound, int upperBound) {
-  return (value >= lowerBound && value <= upperBound);
+  return (static_cast<int>(value) >= lowerBound && static_cast<int>(value) <= upperBound);
```

Comment:
cast lowerbound and upperbound instead

Reply:
DType should be a real number, so it is float

Reply:
In all uses of between function in both files, the argument 'value' takes the values of top_left_x and top_left_y, which are both integer variables. Isn't it? 
DType can take other values yes. But it looks like there's no reason for between function to have the template DType. 
It can just be
```
bool between (int value, int lowerBound, int upperBound)
```

Reply:
@tornadomeet

Reply:
@tornadomeet Is DType template really required for the between functions? As of now it doesn't look like it. Did you use templates for the future?